endorser/api/endpoints/routes/allow.py

```diff
@@ -354,3 +354,78 @@ async def delete_allowed_cred_def(
         return {}
     except Exception as e:
         raise HTTPException(status_code=HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
+
+
+async def update_allowed_config(k, v, db):
+    csvReader = DictReader(iterdecode(k.file, "utf-8"))
+    tmp = {
+        "file_name": k.filename,
+        "contents": [v(**i) for i in csvReader],
+    }
+    for i in tmp["contents"]:
+        db.add(i)
+    return tmp
+
+
+async def update_full_config(
+    publish_did: Optional[UploadFile],
+    schema: Optional[UploadFile],
+    credential_definition: Optional[UploadFile],
+    db: AsyncSession,
+    delete_contents: bool,
+):
+    correlated_tables = {
+        publish_did: AllowedPublicDid,
+        schema: AllowedSchema,
+        credential_definition: AllowedCredentialDefinition,
+    }
+    modifications = {}
+    for k, v in correlated_tables.items():
+        if k:
+            if delete_contents:
+                # TODO look into only deleting after these are stored
```

Comment:
You could look into creating a temporary table with the "old" data and stage everything there until you find that the new data was properly stored/saved.

Reply:
It actually seems that this isn't necessary. I added a test to ensure rollbacks also revert the deletion and it seems to handle that just fine.